### ISSUE TITLE
Does AccuracyMetric compute accumulated or mean values of each batch?

### ISSUE THREAD

**Issue by xiadingZ** [PyTorchLightning/pytorch-lightning#4357]
there are two version of calculate accuracy:
1. calculate accuracy of each batch then do mean operation in `validation_epoch_end`
2. accumulate each batch's `correct, total` values then compute accuracy once in `validation_epoch_end`

which ones does `AccuracyMetric` do?

**Comment by awaelchli** [comment]
The 2nd. 
And it does that also across gpus. 
https://pytorch-lightning.readthedocs.io/en/latest/metrics.html

**Comment by xiadingZ** [comment]
1. In ddp mode, If i want to implement `validation_epoch_end`, how can I get the accumulated accuracy value? below is my current code, but it seems do the 1st method.
```
    def validation_step(self, batch, batch_idx):
        ...
        vid_acc = self.accuracy_video(video_labels_hat, video_labels)
        ...
        return {'val_loss': loss, "val_pacc": part_acc, "val_acc": vid_acc}

    def validation_epoch_end(self, outputs):
        val_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
        val_pacc = torch.stack([x['val_pacc'] for x in outputs]).mean()
        val_acc = torch.stack([x['val_acc'] for x in outputs]).mean()

        dist.all_reduce(val_loss, op=dist.ReduceOp.SUM)
        dist.all_reduce(val_pacc, op=dist.ReduceOp.SUM)
        dist.all_reduce(val_acc, op=dist.ReduceOp.SUM)


        val_loss = val_loss / self.trainer.world_size
        val_pacc = val_pacc / self.trainer.world_size
        val_acc = val_acc / self.trainer.world_size

        self.log('val_loss', val_loss, on_epoch=True, sync_dist=True)
        self.log('val_pacc', val_pacc, on_epoch=True, sync_dist=True)
        self.log('val_acc', val_acc, on_epoch=True, prog_bar=True, sync_dist=True)

        return {'val_loss': val_loss, "val_pacc": val_pacc, "val_acc": val_acc}
```

2. does it mean that I have to create multiple `AccuracyMetric` instances for each variable in each mode, such as `train_vid_acc`, `val_vid_acc`, `test_vid_acc`, `train_img_acc`...?

**Comment by awaelchli** [comment]
You can call .compute() as it is shown in the docs:
https://pytorch-lightning.readthedocs.io/en/latest/metrics.html
Is this what you mean by "accumulated"?

2. Yes you have to create multiple, because they track the internal state of that particular metric.

**Comment by xiadingZ** [comment]
1. yes.  my `accumulated` is what `.compute()` does.   As in docs, it call `self.log` in `training_step` and `training_epoch_end`, but I only call it in `validation_epoch_end`, will this also get the synced value of Metric?

**Comment by SkafteNicki** [comment]
Yes, you can call it in `validation_epoch_end`. When `compute` is called it will sync the Metric, regardless of which method the call is happening in.
